spec/reek/configuration/configuration_file_finder_spec.rb

```diff
@@ -97,6 +97,20 @@
       configuration = described_class.load_from_file(CONFIG_PATH.join('full_mask.reek'))
       expect(configuration).to eq(sample_configuration_loaded)
     end
+
+    context 'with strings for accept and reject settings' do
+      it 'properly converts them' do
+        expected = {
+          'UnusedPrivateMethod' => { 'exclude' => [/exclude regexp/] },
+          'UncommunicativeMethodName' => { 'reject' => [/reject regexp/], 'accept' => [/accept regexp/] },
+          'UncommunicativeModuleName' => { 'reject' => [/reject regexp/], 'accept' => [/accept regexp/] },
+          'UncommunicativeParameterName' => { 'reject' => [/reject regexp/], 'accept' => [/accept regexp/] },
+          'UncommunicativeVariableName' => { 'reject' => [/reject regexp/], 'accept' => [/accept regexp/] }
+        }
+        configuration = described_class.load_from_file(CONFIG_PATH.join('accepts_and_rejects.reek'))
```

Comment:
I think we should specifically document the behavior if people use special regex characters in these strings. Users may  want to match `foo` but not `food`. Can they use `foo$` in this case?

Reply:
>>  Users may want to match foo but not food

The way this works now, no, they can't. Is this a loss? I don't think so. Looking at the way we use disabling smells in our own codebase I can see only 1 case where we actually use regexes:

>> # :reek:UnusedPrivateMethod: { exclude: [ !ruby/regexp /process_/ ] }

and in this case the regex wouldn't have even been necessary.

My point is that while I can see this might be useful in some cases, I think those are edge cases and the majority of users will be fine with us removing this since they don't use it.

I think this is a small price to pay for having a much simpler configuration set up.

Note that you can still achieve this:

>> Users may want to match foo but not food

quite easily - you'll just have to spell it out.

Reply:
Yes :)
With "spelling it out" i meant: If you have "foo" and "food" then "foo" would match both. This definitely might be a huge problem.
Maybe implement something along @chastell idea 

```
I’m not sure how backward-(in)compatible would be to try to guess whether a given string is a name or a regex, but maybe we could just go with a simple heuristic to the tune of strings starting and ending with / are always regexes.
```

right away?

Reply:
Yes, go for it!